provider/ec2/environ.go

```diff
@@ -565,6 +565,17 @@ func (e *environ) StartInstance(ctx context.ProviderCallContext, args environs.S
 	runArgs := commonRunArgs
 	runArgs.AvailZone = availabilityZone
 
+	// TODO (manadart 2020-02-07): We only take the first subnet/zones
+	// mapping to create a NIC for the instance.
+	// This effectively uses a single positive space constraint if many are
+	// specified; behaviour that dates from the original spaces MVP.
+	// It will not take too much effort to enable multi-NIC support for EC2
+	// if we use them all when constructing the instance creation request.
+	var subnetZones map[corenetwork.Id][]string
```

Comment:
Just a note: if anybody does attempt to use this for filling, it will go down panic town...

Reply:
True, but it is only ever read.